src/main/io/serial.c

```diff
@@ -437,9 +432,16 @@ serialPort_t *openSerialPort(
     serialPort_t *serialPort = NULL;
 
 #ifdef USE_SOFTSERIAL
-    if (((identifier == SERIAL_PORT_SOFTSERIAL1) || (identifier == SERIAL_PORT_SOFTSERIAL2)) && (baudRate > 19200)) {
+
+#ifdef USE_OVERRIDE_SOFTSERIAL_BAUDRATE_RESTRICTION
+#define SOFTSERIAL_MAX_BAUDRATE 57600
+#else
+#define SOFTSERIAL_MAX_BAUDRATE 19200
+#endif
+
+    if (((identifier == SERIAL_PORT_SOFTSERIAL1) || (identifier == SERIAL_PORT_SOFTSERIAL2)) && (baudRate > SOFTSERIAL_MAX_BAUDRATE)) {
         // Limit baud rate on soft serial ports
-        baudRate = 19200;
```

Comment:
I don't know. IMO it is better to `return NULL;`. 
Opening port at different baudrate is almost always useless. And port will partilly look like it's working, so it will be even harder to diagnose.
Or you know any case where different baudrate is useful?